tutorials/inner_products/2_physical_properties.py

```diff
@@ -0,0 +1,357 @@
+Ax5.set_title('M-faces Matrix (linear)')
+
+Ax6 = fig.add_subplot(336)
+Ax6.imshow(Mf3.todense())
+Ax6.set_title('M-faces Matrix (anisotropic)')
+
+Ax7 = fig.add_subplot(337)
+Ax7.imshow(Me1.todense())
+Ax7.set_title('M-edges Matrix (isotropic)')
+
+Ax8 = fig.add_subplot(338)
+Ax8.imshow(Me2.todense())
+Ax8.set_title('M-edges Matrix (linear)')
+
+Ax9 = fig.add_subplot(339)
+Ax9.imshow(Me3.todense())
+Ax9.set_title('M-edges Matrix (anisotropic)')
+
+
+#############################################################
+# Spatially Variant Parameters
+# ----------------------------
+#
+# In practice, the parameter :math:`\sigma` or tensor :math:`\Sigma` will
+# vary spatially. In this case, we define the parameter
+# :math:`\sigma` (or parameters :math:`\Sigma`) for each cell. When
+# creating the inner product matrix, we enter these parameters as
+# a numpy array. This is demonstrated below. Properties of the resulting
+# inner product matricies are discussed.
+#
+
+# Create a small 3D mesh
+h = np.ones(5)
+mesh = TensorMesh([h, h, h])
+
+# Isotropic case: (nC, ) numpy array
+sig = np.random.rand(mesh.nC)        # sig for each cell
+Me1 = mesh.getEdgeInnerProduct(sig)  # Edges inner product matrix
+Mf1 = mesh.getFaceInnerProduct(sig)  # Faces inner product matrix
+
+# Linear case: (nC, dim) numpy array
+sig = np.random.rand(mesh.nC, mesh.dim)
+Me2 = mesh.getEdgeInnerProduct(sig)
+Mf2 = mesh.getFaceInnerProduct(sig)
+
+# Anisotropic case: (nC, 3) for 2D and (nC, 6) for 3D
+sig = np.random.rand(mesh.nC, 6)
+Me3 = mesh.getEdgeInnerProduct(sig)
+Mf3 = mesh.getFaceInnerProduct(sig)
+
+# Properties of inner product matricies
+print('\n FACE INNER PRODUCT MATRIX')
+print('- Number of faces              :', mesh.nF)
+print('- Dimensions of operator       :', str(mesh.nF), 'x', str(mesh.nF))
+print('- Number non-zero (isotropic)  :', str(Mf1.nnz))
+print('- Number non-zero (linear)     :', str(Mf2.nnz))
+print('- Number non-zero (anisotropic):', str(Mf3.nnz), '\n')
+
+print('\n EDGE INNER PRODUCT MATRIX')
+print('- Number of faces              :', mesh.nE)
+print('- Dimensions of operator       :', str(mesh.nE), 'x', str(mesh.nE))
+print('- Number non-zero (isotropic)  :', str(Me1.nnz))
+print('- Number non-zero (linear)     :', str(Me2.nnz))
+print('- Number non-zero (anisotropic):', str(Me3.nnz), '\n')
+
+
+#############################################################
+# Inverse
+# -------
+#
+# The final discretized system using the finite volume method may contain
+# the inverse of the inner-product matrix. Here we show how to call this
+# using the *invMat* keyword argument.
+#
+# For the isotropic and linear cases, the inner product matrix is diagonal.
+# As a result, its inverse can be easily formed. For the anisotropic case
+# however, we cannot expicitly form the inverse because the inner product
+# matrix contains a significant number of off-diagonal elements.
+#
+# For the isotropic case, we form M_inv and apply it to a vector using
```

Comment:
Instead of `M_inv`, perhaps `:math:\mathbf{M}^{-1}`?

Reply:
sure